src/Entity/PhoneNumberSettings.php

```diff
@@ -0,0 +1,239 @@
+<?php
```

Comment:
I understand this approach, but I'm still just wondering if we couldn't have achieved a simpler implementation by extending the Drupal core phonenumber field and implementing the additional functionality there? Here we've ended up creating a whole new config entity type just to manage specific settings which could be encapsulated by the field.